spec.md

```diff
@@ -7,9 +7,13 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
 
 ## Distribution Specification
 
-This specification defines an API protocol to facilitate distribution of images.
+This specification defines an API protocol to facilitate distribution of artifacts, including container images.
 
-The goal of this specification is to standardize container image distribution based on the specification for the [Docker Registry HTTP API V2 protocol](https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md).
+The goal of this specification is to standardize artifact distribution based on the specification for the [Docker Registry HTTP API V2 protocol](https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md). 
+
+## OCI Artifact Extensibility 
+
+The distribution spec was initially based on docker images and the [OCI Image spec](https://github.com/opencontainers/image-spec/). As new artifact types surfaced, it became a logical use case to leverage the common elements of distribution to store and retrieve additional artifacts. This spec will reference *images* as one type of artifact, with specifics on how to differentiate different types of artifacts.
```

Comment:
It would be good to explain why it is desirable to store non-image artefacts in a registry. Does this only really make sense for artefacts which reference images?

Reply:
As a spec, I struggled with how much "pitch" to put into the spec, as opposed to speaking to the details. I've written a few different posts, done presentations to set the overall context: 

- [Blog Post: Cloud Native Artifact Registries evolve from Docker Container Registries](https://stevelasker.blog/2019/01/25/cloud-native-artifact-stores-evolve-from-container-registries/)
- [PowerPoint: Leveraging OCI Distribution for new Cloud Native Artifacts](https://github.com/SteveLasker/Presentations/blob/master/ACR/OCIArtifactRegistries.pptx?raw=true)

Do you think we should put more of the pitch here? Are you looking for what type of artifacts to store?

Reply:
> As a spec, I struggled with how much "pitch" to put into the spec, as opposed to speaking to the details. I've written a few different posts, done presentations to set the overall context:
> 
> * [Blog Post: Cloud Native Artifact Registries evolve from Docker Container Registries](https://stevelasker.blog/2019/01/25/cloud-native-artifact-stores-evolve-from-container-registries/)
> * [PowerPoint: Leveraging OCI Distribution for new Cloud Native Artifacts](https://github.com/SteveLasker/Presentations/blob/master/ACR/OCIArtifactRegistries.pptx?raw=true)
> 

Thanks but I don't think it would be appropriate to reference these in the spec.

> Do you think we should put more of the pitch here?

Yes, but not frames as a sales pitch. I'm looking for rationale of when it makes sense to use this feature.

> Are you looking for what type of artifacts to store?

Yes. In particular, does it make any sense to store an artefact that does not reference images?

Reply:
Agree we shouldn't reference these within the spec. I was more providing context on how I was framing things, outside the spec. To provide a funnel of the scenarios, to the **spec**ifics. 

As for when and what artifact types, I don't know if we should attempt to limit. The question I ask in the above [Cloud Native Rejekts talk](https://github.com/SteveLasker/Presentations/blob/master/ACR/OCIArtifactRegistries.pptx?raw=true)  is; ***why would you*** create **Y**et **A**nother **S**torage **S**olution (YASS).

Registries have become ubiquitous. It's hard to see modern workloads that don't use containers. Even IoT devices and cars are running containers because of their packaging, isolation, patching, testing and deployment models. But, lets assume you have an optimized environment that doesn't require container images. Could you justify the engineering effort to build YASS? Or, would you just implement one of the current distribution solutions, and store the artifact types in it?